src/Maths/Silk.NET.Maths/Matrix2x2.cs

```diff
@@ -0,0 +1,458 @@
+        }
+
+        /// <summary>Multiplies a matrix by another matrix.</summary>
+        /// <param name="value1">The first source matrix.</param>
+        /// <param name="value2">The second source matrix.</param>
+        /// <returns>The result of the multiplication.</returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static Matrix2x2<T> Multiply(Matrix2x2<T> value1, Matrix2x2<T> value2) => value1 * value2;
+        
+        /// <summary>Multiplies a matrix by another matrix.</summary>
+        /// <param name="value1">The first source matrix.</param>
+        /// <param name="value2">The second source matrix.</param>
+        /// <returns>The result of the multiplication.</returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static Matrix3x2<T> Multiply(Matrix3x2<T> value1, Matrix2x2<T> value2) => value1 * value2;
+        
+        /// <summary>Multiplies a matrix by another matrix.</summary>
+        /// <param name="value1">The first source matrix.</param>
+        /// <param name="value2">The second source matrix.</param>
+        /// <returns>The result of the multiplication.</returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static Matrix2x3<T> Multiply(Matrix2x2<T> value1, Matrix2x3<T> value2) => value1 * value2;
+
+        /// <summary>Multiplies a matrix by a scalar value.</summary>
+        /// <param name="value1">The source matrix.</param>
+        /// <param name="value2">The scaling factor.</param>
+        /// <returns>The scaled matrix.</returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static Matrix2x2<T> Multiply(Matrix2x2<T> value1, T value2) => value1 * value2;
+        
+        /// <summary>Multiplies a vector by a matrix.</summary>
+        /// <param name="value1">The vector.</param>
+        /// <param name="value2">The matrix.</param>
+        /// <returns>The result of the multiplication.</returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static Vector2<T> Multiply(Vector2<T> value1, Matrix2x2<T> value2)
+            => value1 * value2;
+
+        /// <summary>Returns a new matrix with the negated elements of the given matrix.</summary>
+        /// <param name="value">The source matrix.</param>
+        /// <returns>The negated matrix.</returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static Matrix2x2<T> Negate(Matrix2x2<T> value) => -value;
+
+        /// <summary>Subtracts the second matrix from the first.</summary>
+        /// <param name="value1">The first source matrix.</param>
+        /// <param name="value2">The second source matrix.</param>
+        /// <returns>The result of the subtraction.</returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static Matrix2x2<T> Subtract(Matrix2x2<T> value1, Matrix2x2<T> value2) => value1 - value2;
```

Comment:
Remind me why? I'd argue this makes more sense (because `using Matrix2x2 = Matrix2x2<int>` is then possible)

Reply:
Hmm, I kind of see it. Will try out tomorrow

Reply:
Awesome (it's purely for readability in user code so that they don't have to type out the `<int>` when doing static ops on matrices we know to be ints)